### DISCUSSION TITLE
Will lsp update the initialize method to the newest lsp specification?

### DISCUSSION THREAD

**Discussion by davidmlw** [emacs-lsp/lsp-mode#4083] [General]
Is your feature related or already mentioned on [the wishlist](https://github.com/emacs-lsp/lsp-mode/discussions/2892)?

Describe your feature here.

How about upgrade the initialize workspace with newest LSP specification? Using workspaceFolders to replace rootUri?

https://github.com/emacs-lsp/lsp-mode/blob/05c88ea1bd1ac56bf2a386541047a96240676047/lsp-mode.el#L7628

https://github.com/microsoft/vscode-languageserver-node/blob/2041784436fed53f4e77267a49396bca22a7aacf/protocol/src/common/protocol.ts#L1356-L1363

**Comment by yyoncho** [comment]
lsp-mode will use workspaceFolders if the server is marked as multiroot.

**Comment by davidmlw** [comment]
Thank you very much. 

multiroot is a property of `make-lsp-client`

**Comment by yyoncho** [comment]
https://github.com/search?q=repo%3Aemacs-lsp%2Flsp-mode%20multi-root&type=code
